### DISCUSSION TITLE
batch message to SQS

### DISCUSSION THREAD

**Discussion by sunitagajurel** [celery/kombu#1775] [Q&A]
How can I send batch message to SQS ? I see a method named send_batch_message ,but  I don't know how I can implemet that , sending an array of  dictionaries of message sends that as a single message .

**Comment by TayfurYldz** [comment]
At the Kombu API level, publishing a list of dictionaries is still one message because Producer.publish serializes the object you give it as a single message body. The broker's native batch API is a different operation.

Current Kombu does have a Producer.batch() abstraction, but transport support is explicit: unsupported transports publish immediately inside the context. The current upstream documentation lists the Redis transports as the standard implementations of this batching API; it is not an SQS SendMessageBatch wrapper.

So for SQS specifically, I would not call the boto3 client's send_message_batch through Kombu internals and expect Kombu semantics to remain intact. Kombu's SQS transport is responsible for serialization/envelopes, queue URL resolution, message attributes, and other transport behavior. Bypassing it with raw dictionaries can produce messages that Kombu consumers do not interpret like normal Producer.publish output.

If the requirement is simply to send up to 10 native SQS entries efficiently and the producer/consumer contract is yours, using boto3's SQS SendMessageBatch directly is the clearest choice. If the messages must be normal Kombu messages, use Producer.publish for each one unless/until the SQS transport advertises batch-publish support.

An array passed to publish means one application payload; native SQS batching means several independent SQS messages in one API request. They are different layers and Kombu should only combine them when its SQS transport explicitly implements that mapping.
